test/integration/api/SocialTradingAPI.spec.ts

```diff
@@ -972,6 +974,236 @@ describe('SocialTradingAPI', () => {
+      await subject();
+
+      const poolInfo: any = await socialTradingAPI.fetchTradingPoolTWAPRebalanceDetailsAsync(subjectTradingPool);
+
+      expect(poolInfo.chunkAuctionPeriod).to.be.bignumber.equal(subjectChunkAuctionPeriod);
+    });
+
+    test('successfully updates chunkSize allocation', async () => {
+      await subject();
+
+      const poolInfo: any = await socialTradingAPI.fetchTradingPoolTWAPRebalanceDetailsAsync(subjectTradingPool);
+      const rebalanceVolume = new BigNumber(poolInfo.orderSize).mul(36000).mul(.7);
+      const expectedChunkSize = subjectChunkSize
+                                  .mul(poolInfo.orderSize)
+                                  .div(rebalanceVolume)
+                                  .div(poolInfo.minimumBid).round(0, 3)
+                                  .mul(poolInfo.minimumBid);
+      expect(poolInfo.chunkSize).to.be.bignumber.equal(expectedChunkSize);
+    });
+
+    describe('when the passed allocation is less than 0', async () => {
+      beforeEach(async () => {
+        subjectNewAllocation = new BigNumber(-1);
+      });
+
+      test('throws', async () => {
+        return expect(subject()).to.be.rejectedWith(
+          `The quantity ${subjectNewAllocation.toString()} inputted needs to be greater than zero.`
+        );
+      });
+    });
+
+    describe('when the passed allocation is greater than 100', async () => {
+      beforeEach(async () => {
+        subjectNewAllocation = ether(1.1);
+      });
+
+      test('throws', async () => {
+        return expect(subject()).to.be.rejectedWith(
+          `Provided allocation ${subjectNewAllocation.toString()} is greater than 100%.`
+        );
+      });
+    });
+
+    describe('when the passed allocation is not a multiple of 1%', async () => {
+      beforeEach(async () => {
+        subjectNewAllocation = ether(.012);
+      });
+
+      test('throws', async () => {
+        return expect(subject()).to.be.rejectedWith(
+          `Provided allocation ${subjectNewAllocation.toString()} is not multiple of 1% (10 ** 16)`
+        );
+      });
+    });
+
+    describe('when the caller is not the trader', async () => {
+      beforeEach(async () => {
+        subjectCaller = ACCOUNTS[3].address;
+      });
+
+      afterEach(async () => {
+        subjectCaller = DEFAULT_ACCOUNT;
+      });
+
+      test('throws', async () => {
+        return expect(subject()).to.be.rejectedWith(
+          `Caller ${subjectCaller} is not trader of tradingPool.`
+        );
+      });
+    });
+
+    describe('when passed chunkSize is outside of bounds', async () => {
+      beforeEach(async () => {
+        subjectChunkSize = ether(100);
+      });
+
+      test('throws', async () => {
+        return expect(subject()).to.be.rejectedWith(
+          `Passed chunk size must be between ${assetPairBounds[0].bounds.lower} and` +
+          ` ${assetPairBounds[0].bounds.upper}.`
+        );
+      });
+    });
```

Comment:
Not sure how `chunkAuctionPeriod` factors into this. I'll throw out having a test case for invalid `chunkAuctionPeriod` values

Reply:
Suppose we could test to make sure it's greater than 0

Reply:
I guess it depends how thorough we want to make our checks here. Unfortunately there's a lot of calls to the chain that need to be made in order to do some of these checks

Reply:
Sure you guys know better than I, just throwing out a suggestion